torch/_refs/__init__.py

```diff
@@ -3320,11 +3316,9 @@ def new_full(
     device: Optional[torch.device] = None,
     pin_memory: bool = False,
 ) -> TensorLikeType:
-    r = a.new_empty(
-        size, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
+    return torch.full(
+        size, fill_value, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
```

Comment:
you are incorrectly setting dtype and device - in case they are not supplied, they are taken from `a`. Test errors are real.

Reply:
fixed.